core/src/main/java/org/elasticsearch/action/get/GetResponse.java

```diff
@@ -160,6 +161,22 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     public static GetResponse fromXContent(XContentParser parser) throws IOException {
         GetResult getResult = GetResult.fromXContent(parser);
+
+        // At this stage we ensure that we parsed enough information to return
+        // a valid GetResponse instance. If it's not the case, we throw an
+        // exception so that callers know it and can handle it correctly.
+        // This is typically the case when one wants to parse the result of
```

Comment:
this is the case when we get a 404 back, which can be parsed as a normal get response with found set to false, or as an elasticsearch exception. The caller of this method needs a way to figure out whether we got back a valid get response, which can be done by catching ParsingException.

Maybe some of this should be in the javadocs of this method.